src/services_models/GroupReserve.jl

```diff
@@ -0,0 +1,88 @@
+struct GroupReserve <: AbstractReservesFormulation end
+############################### Reserve Variables` #########################################
+"""
+This function checks if the variables for reserves were created
+"""
+function check_activeservice_variables(
+    psi_container::PSIContainer,
+    contributing_services::Vector{<:PSY.Service},
+)
+    for service in contributing_services
+        # Should pop an error if no such variable exists
+        reserve_variable =
+            get_variable(psi_container, PSY.get_name(service), typeof(service))
+    end
+    return
+end
+
+################################## Reserve Requirement Constraint ##########################
+"""
+This function creates the requirement constraint that will be attained by the apropriate services
+"""
+function service_requirement_constraint!(
+    psi_container::PSIContainer,
+    service::SR,
+    ::ServiceModel{SR, GroupReserve},
+    contributing_services::Vector{<:PSY.Service},
+) where {SR <: PSY.StaticReserveGroup}
+    parameters = model_has_parameters(psi_container)
+    use_forecast_data = model_uses_forecasts(psi_container)
+    initial_time = model_initial_time(psi_container)
+    @debug initial_time
+    time_steps = model_time_steps(psi_container)
+    name = PSY.get_name(service)
+    constraint = get_constraint(psi_container, make_constraint_name(REQUIREMENT, SR))
+    use_slacks = get_services_slack_variables(psi_container.settings)
+    reserve_variables = [
+        get_variable(psi_container, PSY.get_name(r), typeof(r))
+        for r in contributing_services
+    ]
+
+    if use_forecast_data && :forecast ∈ fieldnames(typeof(service))
```

Comment:
may i ask about the use of `:forecast ∈ fieldnames(...)` in this PR? What is the role it is playing?
is it being used as a proxy for some train like "has forecasts"? I wonder if it might just be better to have a function for that? 

Also, AFAICT `fieldnames` is not used in this way anywhere else in PSim, and usually field names are not considered public API (so given `Service` comes from PSys should perhaps not be relied upon here?)

Reply:
Good point! just changed to has_forecasts.